File: app/src/main/java/com/durbar/bangabandhuplay/data/repository/MoreHomeRepository.java
package com.durbar.bangabandhuplay.data.repository;

import android.app.Application;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.lifecycle.MutableLiveData;

import com.durbar.bangabandhuplay.data.ApiService;
import com.durbar.bangabandhuplay.data.Api;
import com.durbar.bangabandhuplay.data.model.category.single_sub.SingleSubCategoryRes;
import com.durbar.bangabandhuplay.data.model.frontend_custom_content.custom_content_by_slug.CustomContentBySlug;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MoreHomeRepository {
    private Application application;
    private MutableLiveData<CustomContentBySlug> contentHome;
    private MutableLiveData<SingleSubCategoryRes> singleSubCategory;
    private ApiService apiService;

    public MoreHomeRepository(Application application) {
        this.application = application;
        apiService = Api.getInstance().getApiService();
        contentHome = new MutableLiveData<>();
        singleSubCategory = new MutableLiveData<>();
    }

    public MutableLiveData<CustomContentBySlug> fetchContentForHome(String slug) {
        Call<CustomContentBySlug> call = apiService.getCustomContentBySlug(slug);
        call.enqueue(new Callback<CustomContentBySlug>() {
            @Override
            public void onResponse(Call<CustomContentBySlug> call, @NonNull Response<CustomContentBySlug> response) {
                if (response.isSuccessful() && response.body() != null) {
                    //Log.v(TAG, response.body().getMessage());
                    contentHome.setValue(response.body());
                } else {
                    Toast.makeText(application, "Something went wrong!!" + response.code(), Toast.LENGTH_SHORT).show();
                    //Log.v(TAG, "response not successful");
                }
            }

            @Override
            public void onFailure(Call<CustomContentBySlug> call, Throwable t) {
                Toast.makeText(application, "Failed to connect", Toast.LENGTH_SHORT).show();
                t.printStackTrace();
            }
        });
        return contentHome;
    }

    public MutableLiveData<SingleSubCategoryRes> getSingleSubCategoryContents(String slug){
        Call<SingleSubCategoryRes> call = apiService.getSingleSubCategoryContents(slug);
        call.enqueue(new Callback<SingleSubCategoryRes>() {
            @Override
            public void onResponse(Call<SingleSubCategoryRes> call, Response<SingleSubCategoryRes> response) {
                if (response.isSuccessful() && response.body() != null) {
                    singleSubCategory.setValue(response.body());
                } else {
                    Toast.makeText(application, "Something went wrong!!" + response.code(), Toast.LENGTH_SHORT).show();
                }
            }

            @Override
            public void onFailure(Call<SingleSubCategoryRes> call, Throwable t) {
                Toast.makeText(application, "Failed to connect", Toast.LENGTH_SHORT).show();
                t.printStackTrace();
            }
        });
        return singleSubCategory;
    }
}

File: app/src/main/java/com/durbar/bangabandhuplay/data/repository/MoreHomeRepository.kt
package com.durbar.bangabandhuplay.data.repository

import android.app.Application
import android.widget.Toast
import androidx.lifecycle.MutableLiveData
import com.durbar.bangabandhuplay.data.Api
import com.durbar.bangabandhuplay.data.ApiService
import com.durbar.bangabandhuplay.data.model.category.single_sub.SingleSubCategoryRes
import com.durbar.bangabandhuplay.data.model.frontend_custom_content.custom_content_by_slug.CustomContentBySlug
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MoreHomeRepository(private val application: Application) {
    private val contentHome: MutableLiveData<CustomContentBySlug?>
    private val singleSubCategory: MutableLiveData<SingleSubCategoryRes?>
    private val apiService: ApiService

    init {
        apiService = Api.instance!!.apiService
        contentHome = MutableLiveData()
        singleSubCategory = MutableLiveData()
    }

    fun fetchContentForHome(slug: String?): MutableLiveData<CustomContentBySlug?> {
        val call = apiService.getCustomContentBySlug(slug)
        call.enqueue(object : Callback<CustomContentBySlug?> {
            override fun onResponse(
                call: Call<CustomContentBySlug?>,
                response: Response<CustomContentBySlug?>
            ) {
                if (response.isSuccessful && response.body() != null) {
                    //Log.v(TAG, response.body().getMessage());
                    contentHome.setValue(response.body())
                } else {
                    Toast.makeText(
                        application,
                        "Something went wrong!!" + response.code(),
                        Toast.LENGTH_SHORT
                    ).show()
                    //Log.v(TAG, "response not successful");
                }            }

            override fun onFailure(call: Call<CustomContentBySlug?>, t: Throwable) {
                Toast.makeText(application, "Failed to connect", Toast.LENGTH_SHORT).show()
                t.printStackTrace()          }
        })
        return contentHome
    }

    fun getSingleSubCategoryContents(slug: String?): MutableLiveData<SingleSubCategoryRes?> {
        val call = apiService.getSingleSubCategoryContents(slug)
        call.enqueue(object : Callback<SingleSubCategoryRes?> {
            override fun onResponse(
                call: Call<SingleSubCategoryRes?>,
                response: Response<SingleSubCategoryRes?>
            ) {
                if (response.isSuccessful && response.body() != null) {
                    singleSubCategory.setValue(response.body())
                } else {
                    Toast.makeText(
                        application,
                        "Something went wrong!!" + response.code(),
                        Toast.LENGTH_SHORT
                    ).show()
                }            }

            override fun onFailure(call: Call<SingleSubCategoryRes?>, t: Throwable) {
                Toast.makeText(application, "Failed to connect", Toast.LENGTH_SHORT).show()
                t.printStackTrace()            }
        })
        return singleSubCategory
    }
}
